### PR TITLE
Add missing commit states to PR checks template

### PR DESCRIPTION
**Pull request by CirnoT**
The template was missing case for `failure` state, causing 'Some checks are pending' to be displayed incorrectly.

There was also a missing case for `warning` - I have decided to use translation for `success` in such case, but we might want to add new translation for it.

---
On an unrelated note, and possibly for another PR; we might want to distinguish between `error` and `failure` state in translations, just like GitHub does (errored vs failed) as they can mean very different things (for example CI failed to execute pipeline due to some internal error vs pipeline failed)

### CODE REVIEW

File: templates/repo/pulls/status.tmpl [hunk 1]
```diff
@@ -4,6 +4,10 @@
             {{$.i18n.Tr "repo.pulls.status_checking"}}
         {{else if eq .LatestCommitStatus.State "success"}}
             {{$.i18n.Tr "repo.pulls.status_checks_success"}}
+        {{else if eq .LatestCommitStatus.State "warning"}}
+            {{$.i18n.Tr "repo.pulls.status_checks_success"}}
```
Comment (6543) [inline templates/repo/pulls/status.tmpl]: can you add a new string to translation for this?

**Comment by CirnoT** [comment]
Added translation string for `warning` as per review.

Took liberty of using this opportunity to separate `failure` and `error` as well, since they are differentiated by icons everywhere else.

**Review by 6543 — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by jolheiser** [comment]
Ping lgtm

**Comment by lunny** [comment]
Shouldn't this be backport to v1.11 ?

**Comment by lafriks** [comment]
It is just display issue, not critical bug

**Pull request merged**